<a href="https://colab.research.google.com/github/tanush-47/GrayScaleVideoConverter/blob/main/GrayScaleVideoConverter.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [8]:
%pip install opencv-python

## Download a video

### Subtask:
Download a sample video file from the internet.


In [9]:
import requests

video_url = "https://test-videos.co.uk/vids/bigbuckbunny/mp4/720/Big_Buck_Bunny_720p_1mb.mp4"
local_video_path = "sample_video.mp4"

try:
    response = requests.get(video_url, stream=True)
    response.raise_for_status()  # Raise an exception for bad status codes

    with open(local_video_path, 'wb') as f:
        for chunk in response.iter_content(chunk_size=8192):
            f.write(chunk)

    print(f"Video downloaded successfully to {local_video_path}")

except requests.exceptions.RequestException as e:
    print(f"Error downloading video: {e}")
    download_successful = False
else:
    download_successful = True

Error downloading video: 404 Client Error: Not Found for url: https://test-videos.co.uk/vids/bigbuckbunny/mp4/720/Big_Buck_Bunny_720p_1mb.mp4


**Reasoning**:
The previous download attempts failed due to 404 errors. Try yet another publicly available video URL to download the sample video file and save it locally.



In [10]:
video_url = "https://sample-videos.com/video123/mp4/720/big_buck_bunny_720p_1mb.mp4"
local_video_path = "sample_video.mp4"

try:
    response = requests.get(video_url, stream=True)
    response.raise_for_status()  # Raise an exception for bad status codes

    with open(local_video_path, 'wb') as f:
        for chunk in response.iter_content(chunk_size=8192):
            f.write(chunk)

    print(f"Video downloaded successfully to {local_video_path}")

except requests.exceptions.RequestException as e:
    print(f"Error downloading video: {e}")
    download_successful = False
else:
    download_successful = True

Error downloading video: 404 Client Error: Not Found for url: https://sample-videos.com/video123/mp4/720/big_buck_bunny_720p_1mb.mp4


## Download a video

### Subtask:
Download a sample video file from the internet.


**Reasoning**:
Attempt to download a sample video from a known working URL using requests and handle potential errors.



In [11]:
import requests

video_url = "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"
local_video_path = "sample_video.mp4"

try:
    response = requests.get(video_url, stream=True)
    response.raise_for_status()

    with open(local_video_path, 'wb') as f:
        for chunk in response.iter_content(chunk_size=8192):
            f.write(chunk)

    print(f"Video downloaded successfully to {local_video_path}")
    download_successful = True

except requests.exceptions.RequestException as e:
    print(f"Error downloading video: {e}")
    download_successful = False

Video downloaded successfully to sample_video.mp4


In [12]:
import cv2

if not download_successful:
    print("Video was not downloaded successfully. Skipping grayscale conversion.")
else:
    local_video_path = "sample_video.mp4"
    grayscale_video_path = "grayscale_video.mp4"

    cap = cv2.VideoCapture(local_video_path)

    if not cap.isOpened():
        print(f"Error: Could not open video file {local_video_path}")
    else:
        fps = int(cap.get(cv2.CAP_PROP_FPS))
        width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
        height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))

        fourcc = cv2.VideoWriter_fourcc(*'mp4v')
        out = cv2.VideoWriter(grayscale_video_path, fourcc, fps, (width, height), isColor=False)

        while cap.isOpened():
            ret, frame = cap.read()
            if ret:
                gray_frame = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
                out.write(gray_frame)
            else:
                break

        cap.release()
        out.release()
        print(f"Grayscale video saved successfully to {grayscale_video_path}")

Grayscale video saved successfully to grayscale_video.mp4


**Reasoning**:
The previous download attempts failed due to 404 errors. Try yet another publicly available video URL to download the sample video file and save it locally.

In [13]:
import requests

video_url = "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"
local_video_path = "sample_video.mp4"

try:
    response = requests.get(video_url, stream=True)
    response.raise_for_status()

    with open(local_video_path, 'wb') as f:
        for chunk in response.iter_content(chunk_size=8192):
            f.write(chunk)

    print(f"Video downloaded successfully to {local_video_path}")
    download_successful = True

except requests.exceptions.RequestException as e:
    print(f"Error downloading video: {e}")
    download_successful = False

Video downloaded successfully to sample_video.mp4


Now that the video is downloaded, convert it to grayscale using OpenCV.

In [14]:
import cv2

if not download_successful:
    print("Video was not downloaded successfully. Skipping grayscale conversion.")
else:
    local_video_path = "sample_video.mp4"
    grayscale_video_path = "grayscale_video.mp4"

    cap = cv2.VideoCapture(local_video_path)

    if not cap.isOpened():
        print(f"Error: Could not open video file {local_video_path}")
    else:
        fps = int(cap.get(cv2.CAP_PROP_FPS))
        width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
        height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))

        fourcc = cv2.VideoWriter_fourcc(*'mp4v')
        out = cv2.VideoWriter(grayscale_video_path, fourcc, fps, (width, height), isColor=False)

        while cap.isOpened():
            ret, frame = cap.read()
            if ret:
                gray_frame = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
                out.write(gray_frame)
            else:
                break

        cap.release()
        out.release()
        print(f"Grayscale video saved successfully to {grayscale_video_path}")

Grayscale video saved successfully to grayscale_video.mp4
